app/controllers/evidence_controller.rb

```diff
@@ -62,7 +63,15 @@ def return_letter
   end
 
   def return_application
-    redirect_to root_path if ResolverService.new(evidence, current_user).return
+    if ResolverService.new(evidence, current_user).return
+      process_evidence_check_flag
+      redirect_to root_path
+    end
+  end
+
+  def process_evidence_check_flag
```

Comment:
I would add the fact that the service is called from `summary_save` and `return_application` to the controller spec. It doesn't really change the flow of the controller, but it's important to make sure this line is not deleted.